ledger-service/cli-opts/src/main/scala/cliopts/package.scala

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml
+
+package object cliopts {
+
+  /** A lens-style setter.  When you want nested structures to be possible, this
+    * is vastly superior to the more obvious `(B, T) => T`, because unlike that
+    * one, this form permits nesting via trivial composition.
+    */
+  type Setter[T, B] = (B => B, T) => T
```

Comment:
Why is this `(B => B, T) => T` but not `(Unit => B, T) => T`? Or is that in case you want to have access to the default value that is already present in the config and being overwritten? We currently don't use the passed values, we throw it away, so it can be `() => B`:
```
.action((x, c) => address(_ => x, c))
```
See the underscore in `_ => x`.

Reply:
This is explained in the Scaladoc; we do not throw it away in every case. With `Unit` there, it is impossible to compose a `Setter[T, U]` and `Setter[U, V]` to get a `Setter[T, V]`; `TlsConfigurationCli` uses this feature.

The alternative is to have two `Setter` types.

Reply:
actually yeah, it is explained in the scaladoc... but who reads them :)